libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -92,7 +92,8 @@ internal class MapboxRerouteController @VisibleForTesting constructor(
             val builder = toBuilder()
 
             if (this.profile() == DirectionsCriteria.PROFILE_DRIVING ||
-                this.profile() == DirectionsCriteria.PROFILE_DRIVING_TRAFFIC
+                this.profile() == DirectionsCriteria.PROFILE_DRIVING_TRAFFIC ||
+                this.profile().endsWith("-EXPERIMENTAL")
```

Comment:
This is super duper not obvious. `profile()` is marked with the `@DirectionsCriteria.ProfileCriteria` annotation.
This additional check would blow anyone's mind.
I know you said it was temporary. But we still need to support it client-side. Let's support it properly: either remove the annotation or add this specific profile to the list.

Reply:
I we don't have to merge it to main and support it. If the plain is to release one or two snapshots for the experiment, I'm okay with the code which is not obvious. If we need a better support for the main branch, we can implement it in a different way.

Reply:
If we don't merge it to main, I'm absolutely OK with this approach. Unless we expect any other similar experiments in the future.